rust/catalyst-types/src/mmap_file.rs

```diff
@@ -0,0 +1,193 @@
+//! Memory-mapped file.
+
+use core::fmt;
+use std::{
+    path::Path,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+
+use fmmap::{MmapFile, MmapFileExt};
+use once_cell::sync::Lazy;
+use serde::{ser::SerializeStruct, Serialize, Serializer};
+
+/// Memory-mapped file.
+pub struct MemoryMapFile {
+    /// The memory-mapped file.
+    file: MmapFile,
+    /// The size of the memory-mapped file.
+    size: u64,
+}
+/// Global statistic for memory-mapped files.
+static MEMMAP_FILE_STATS: Lazy<MemMapFileStat> = Lazy::new(MemMapFileStat::default);
+
+/// Memory-mapped file statistic.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct MemMapFileStat(Arc<MemMapFileStatInner>);
```

Comment:
We don't need multiple copies of this, we should just have one internally.
Is there any other reason for the Arc?

This also means you don't need the inner.